### PR TITLE
DAOS-11077 DFS: change mtime/atime/ctime in inode to store hlc

### PR DESCRIPTION
**Pull request by mchaarawi**
Using HLC converted from timespec allows for nsec precision.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

**Review by jolivier23 — APPROVED** [review]

File: src/client/dfs/dfs.c [hunk 1]
```diff
@@ -63,9 +63,9 @@
 #define MODE_IDX	0
 #define OID_IDX		(sizeof(mode_t))
 #define ATIME_IDX	(OID_IDX + sizeof(daos_obj_id_t))
-#define MTIME_IDX	(ATIME_IDX + sizeof(time_t))
-#define CTIME_IDX	(MTIME_IDX + sizeof(time_t))
-#define CSIZE_IDX	(CTIME_IDX + sizeof(time_t))
+#define MTIME_IDX	(ATIME_IDX + sizeof(uint64_t))
+#define CTIME_IDX	(MTIME_IDX + sizeof(uint64_t))
+#define CSIZE_IDX	(CTIME_IDX + sizeof(uint64_t))
```
Comment (daltonbohning) [inline src/client/dfs/dfs.c:68]: This implies this works as expected, right?
```
time_t mtime = ...;
uint64_t mtime_hlc = (uint64_t) mtime;
```
Meaning, if we read an older container that stored times as `time_t`, but into a `uint64_t`, it's the same time?

Comment (ashleypittman) [reply src/client/dfs/dfs.c:68]: The suggestion is we squash the two PRs at/before landing so there is no "older" container.

Comment (daltonbohning) [reply src/client/dfs/dfs.c:68]: What about a 2.0 container?

Comment (ashleypittman) [reply src/client/dfs/dfs.c:68]: Reviewing this now, the key is in entry_stat in https://github.com/daos-stack/daos/pull/9636/files

There is code there to read the values the old way if there is no mtime handling so I think it's sufficient that these types are the correct size.

File: src/client/dfs/dfs.c [hunk 2]
```diff
@@ -1532,7 +1573,13 @@ dfs_cont_create(daos_handle_t poh, uuid_t *cuuid, dfs_attr_t *attr,
 	/** Add root object */
 	entry.oid = roots.cr_oids[1];
 	entry.mode = S_IFDIR | 0755;
-	entry.atime = entry.mtime = entry.ctime = time(NULL);
+	rc = clock_gettime(CLOCK_REALTIME, &now);
+	if (rc)
+		D_GOTO(err_super, rc = errno);
+	rc = crt_timespec2hlc(now, &hlc);
+	if (rc)
+		D_GOTO(err_super, rc = daos_der2errno(rc));
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/client/dfs/dfs.c:1581]: There is some replication of code there but no, those functions do different things.

**Review by daltonbohning — APPROVED** [review]

File: src/client/dfs/dfs.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfs/dfs.c:68]: I have double-checked now and yes they are.

**Comment by daltonbohning** [comment]
> Nacking this PR just so it doesn't get landed accidentally.

We could also just set it to draft

**Pull request closed without merging**